CHANGES.rst

```diff
@@ -1,3 +1,23 @@
+2.1.0 (2023-11-08)
+==================
+
```

Comment:
Since this changelog entry might be getting a lot of eyeballs, we should maybe move the migration guide link again up here?